src/providers/provider.ts

```diff
@@ -17,13 +18,31 @@ export class Provider {
   protected _eventManager: EventManager;
   protected _logger: Logger;
   public cuePointManager: CuePointManager | null = null;
+  private _dataCollector: DataCollector | null = null;
 
   constructor(player: Player, eventManager: EventManager, logger: Logger, types: CuepointTypeMap) {
     this._types = types;
     this._logger = logger;
     this._player = player;
     this._eventManager = eventManager;
     this._logger = logger;
+    if (this._useDataCollector()) {
+      // for live entry without DVR use additional processing of cues (filter out cues behind Live Edge)
+      const onTimeoutFn = (collectedData: CuePoint[]) => {
+        // filter out duplicates
+        const collectedDataMap = new Map();
+        collectedData.forEach(cuePoint => {
```

Comment:
can you use a Set to filter duplicates or does Set not help in this case ?

Reply:
collectedData has next view:
[
 {id: 1, startTime: x1, endTime: infinite},
 {id: 1, startTime: x1, endTime: x1},
 {id: 2, startTime: x2, endTime: infinite},
 ... etc
]
co firstly we have to get latest item (by id) and only then make filtering